src/libtiled/mapreader.cpp

```diff
@@ -834,6 +873,48 @@ void MapReaderPrivate::readImageLayerImage(ImageLayer &imageLayer)
     xml.skipCurrentElement();
 }
 
+inline void MapReaderPrivate::readObjectData(MapObject *object)
+{
+    while (xml.readNextStartElement()) {
+        if (xml.name() == QLatin1String("properties")) {
+            object->mergeProperties(readProperties());
+        } else if (xml.name() == QLatin1String("polygon")) {
+            object->setPolygon(readPolygon());
+            object->setShape(MapObject::Polygon);
+        } else if (xml.name() == QLatin1String("polyline")) {
+            object->setPolygon(readPolygon());
+            object->setShape(MapObject::Polyline);
+        } else if (xml.name() == QLatin1String("ellipse")) {
+            xml.skipCurrentElement();
+            object->setShape(MapObject::Ellipse);
+        } else if (xml.name() == QLatin1String("text")) {
+            object->setTextData(readObjectText());
+            object->setShape(MapObject::Text);
+        } else {
+            readUnknownElement();
+        }
+    }
+}
+
+MapObject *MapReaderPrivate::readTemplate()
+{
+    Q_ASSERT(xml.isStartElement() && xml.name() == QLatin1String("template"));
+    const QXmlStreamAttributes atts = xml.attributes();
+    const QString name = atts.value(QLatin1String("name")).toString();
+//    const unsigned gid = atts.value(QLatin1String("gid")).toUInt();
+    const qreal width = atts.value(QLatin1String("width")).toDouble();
+    const qreal height = atts.value(QLatin1String("height")).toDouble();
+    const QString type = atts.value(QLatin1String("type")).toString();
+
+    const QSizeF size(width, height);
+
+    MapObject *object = new MapObject(name, type, {0, 0}, size);
+
+    readObjectData(object);
```

Comment:
I think it would be fine to prevent the writing of the ID by just making sure it is 0 and adding a check for this in `MapWriterPrivate::writeObject`. For now, I would not bother with preventing the writing of the position.